baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/safety/SafetyPropagationTransfer.java

```diff
@@ -202,6 +212,28 @@
                     .onDescendantOf(OptionalDouble.class.getName())
                     .namedAnyOf("getAsDouble", "orElseThrow"));
 
+    private static final Matcher<ExpressionTree> STREAM_ACCESSORS = Matchers.anyOf(
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(BaseStream.class.getName())
+                    .namedAnyOf("iterator", "spliterator", "sequential", "parallel", "unordered", "onClose"),
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Stream.class.getName())
+                    .namedAnyOf(
+                            "filter",
+                            "distinct",
+                            "sorted",
+                            "peek",
+                            "limit",
+                            "skip",
+                            "takeWhile",
+                            "dropWhile",
+                            "toArray",
+                            "toList",
+                            "min",
+                            "max",
+                            "findFirst",
+                            "findAny"));
```

Comment:
Shouldn't this be every instance method on `Stream` that returns another `Stream`? Maybe we can list those methods using reflection instead of listing them manually?

Reply:
Not quite, the list does not include `map`/`flatMap`/`mapTo*` because those methods may modify the safety of a stream. e.g. `streamOfUnsafeObject.map(unsafeObject -> unsafeObject.getSafeField())` is safe. Lambda handling for this sort of thing is super duper tough though, and I'm still trying to figure out the best way to do it.